multiqc/modules/xenome/xenome.py

```diff
@@ -0,0 +1,124 @@
+        )
+
+        # Parse logs
+        # Find and load any Xenome reports
+        self.xenome_data = dict()
+        for f in self.find_log_files("xenome"):
+            self.parse_xenome_logs(f)
+
+        # Filter to strip out ignored sample names
+        self.xenome_data = self.ignore_samples(self.xenome_data)
+
+        if len(self.xenome_data) == 0:
+            raise UserWarning
+
+        log.info("Found {} reports".format(len(self.xenome_data)))
+
+        # Write parsed report data to a file
+        self.write_data_file(self.xenome_data, "multiqc_xenome")
+
+        # Basic Stats Table
+        # Report table is immutable, so just updating it works
+        self.xenome_general_stats_table()
+
+        # Alignment Rate Plot
+        self.xenome_stats_plot()
+
+    def parse_xenome_logs(self, f):
+        s_name = f["s_name"].replace("_xenome_stats", "")
+        parsed_data = {}
+        regexes = {
+            "human_reads": r"(\d+)\s+\d+.\d+\s+human",
+            "mouse_reads": r"(\d+)\s+\d+.\d+\s+mouse",
+            "both_human_mouse_reads": r"(\d+)\s+\d+.\d+\s+both",
+            "neither_human_mouse_reads": r"(\d+)\s+\d+.\d+\s+neither",
+            "ambiguous_reads": r"(\d+)\s+\d+.\d+\s+ambiguous",
+        }
+
+        for l in f["f"].splitlines():
+            for k, r in regexes.items():
+                match = re.search(r, l)
+                if match:
+                    parsed_data[k] = float(match.group(1))
+
+        if len(parsed_data) > 0:
+            if s_name in self.xenome_data:
+                log.debug("Duplicate sample name found! Overwriting: {}".format(s_name))
+            self.add_data_source(f, s_name)
+            self.xenome_data[s_name] = parsed_data
+
+    def xenome_general_stats_table(self):
+        """Take the parsed stats from the Xenome log and add it to the
+        basic stats table at the top of the report"""
+
+        headers = OrderedDict()
+        headers["human_reads"] = {
+            "title": "Number Human Reads",
+            "description": "The number of human reads in the sample",
+            "min": 0,
+            "scale": False,
+        }
+        headers["mouse_reads"] = {
+            "title": "Number Mouse Reads",
+            "description": "The number of human reads in the sample",
+            "min": 0,
+            "scale": False,
+        }
+        self.general_stats_addcols(self.xenome_data, headers)
+
+    def xenome_stats_plot(self):
+        """Make the HighCharts HTML to plot the classification rates"""
+
+        # Specify the order of the different possible categories
+        keys = OrderedDict()
+        keys["human_reads"] = {"color": "#000034", "name": "Human"}
+        keys["mouse_reads"] = {"color": "#328AE2", "name": "Mouse"}
+        keys["both_human_mouse_reads"] = {"color": "#7F9DA7", "name": "Both Species"}
+        keys["neither_human_mouse_reads"] = {"color": "#C1B8AA", "name": "Neither Species"}
```

Comment:
I think the tool can be run on two arbitrary fastq files, which can belong to other species than Human and Mouse. Do you think it makes sense to generalise the columns, and call them Species 1 and Species 2? Or somehow parse the titles from the logs?

Also, see the [disambiguate](https://github.com/ewels/MultiQC/blob/master/multiqc/modules/disambiguate/disambiguate.py) module. Not sure if it's a reference implementation - maybe it can improved as well - but it would be good to make the outputs for both modules look similar in terms of colors, scales, etc.

Reply:
I think if it's available in the logs/outputs, it can be captured; otherwise, it can be by default "Species a" and "Species b" (like in disambiguate), and customised via a MultiQC config (similarly like there is a special [config section for qualimap](https://github.com/ewels/MultiQC/blob/8e471b39740ff8e1b31b499f9b7ba8ba42af6ed2/docs/modules/qualimap.md?plain=1#L41C1-L53))

Reply:
Regarding color matching among modules: when I make plots, I avoid using forms red/green in plots wherever possible. Diambiguate uses `#B1084C`, which I would prefer not to use. If you would like me to match though, I can implement that. The tool methods do differ in that disambiguate is based on mapped reads, whereas xenome uses a kmer approach to bin reads into either host or graft without mapping. 

Solving the flexibility issue is going to take me time. Being able to grep on a string was easy enough for me to code up. Matching strings based on physical position in the log file is a bit beyond me at the moment.  Hopefully I can find time next week

Reply:
In terms of color matching, we can instead adjust Diambiguate, of the red color doesn't make sense here.